Objects/unicodeobject.c

```diff
@@ -12505,11 +12549,47 @@ Return -1 on failure.
 [clinic start generated code]*/
 
 static Py_ssize_t
-unicode_rfind_impl(PyObject *str, PyObject *substr, Py_ssize_t start,
+unicode_rfind_impl(PyObject *str, PyObject *subobj, Py_ssize_t start,
                    Py_ssize_t end)
-/*[clinic end generated code: output=880b29f01dd014c8 input=898361fb71f59294]*/
+/*[clinic end generated code: output=9d316eee7b9f9bf0 input=23ae7964e8f70b35]*/
 {
-    Py_ssize_t result = any_find_slice(str, substr, start, end, -1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        Py_ssize_t len = PyUnicode_GET_LENGTH(str);
+        ADJUST_INDICES(start, end, len);
+        // Work in batches of 10000
+        for (; result == -1 && end >= start; end -= 10000) {
```

Comment:
Should it be `>`? If yes, same for bytes?

Reply:
No, then an empty substring won't match the empty string.